openedx/core/djangoapps/user_authn/views/tests/test_login.py

```diff
@@ -482,6 +487,45 @@ def test_check_password_policy_compliance_warning(self):
             self.assertIn('Test warning', self.client.session['_messages'])
         self.assertTrue(response_content.get('success'))
 
+    @ddt.data(
+        ('test_password', 'test_password', True),
+        (unicodedata.normalize('NFKD', u'Ṗŕệṿïệẅ Ṯệẍt'),
+         unicodedata.normalize('NFKC', u'Ṗŕệṿïệẅ Ṯệẍt'), False),
+        (unicodedata.normalize('NFKC', u'Ṗŕệṿïệẅ Ṯệẍt'),
+         unicodedata.normalize('NFKD', u'Ṗŕệṿïệẅ Ṯệẍt'), True),
+        (unicodedata.normalize('NFKD', u'Ṗŕệṿïệẅ Ṯệẍt'),
+         unicodedata.normalize('NFKD', u'Ṗŕệṿïệẅ Ṯệẍt'), False),
+    )
+    @ddt.unpack
+    @override_waffle_flag(PASSWORD_UNICODE_NORMALIZE, active=True)
+    def test_password_unicode_normalization_login(self, password, password_entered, login_success):
+        """
+        Tests unicode normalization on user's passwords on login.
+        """
+        self.user.set_password(password)
+        self.user.save()
+        response, mock_audit_log = self._login_response(self.user.email, password_entered,)
+        self._assert_response(response, success=login_success)
+
+    @ddt.data(
+        ('test_password', 'test_password', True),
+        (unicodedata.normalize('NFKD', u'Ṗŕệṿïệẅ Ṯệẍt'),
+         unicodedata.normalize('NFKC', u'Ṗŕệṿïệẅ Ṯệẍt'), False),
+        (unicodedata.normalize('NFKC', u'Ṗŕệṿïệẅ Ṯệẍt'),
+         unicodedata.normalize('NFKD', u'Ṗŕệṿïệẅ Ṯệẍt'), False),
+        (unicodedata.normalize('NFKD', u'Ṗŕệṿïệẅ Ṯệẍt'),
+         unicodedata.normalize('NFKD', u'Ṗŕệṿïệẅ Ṯệẍt'), True),
+    )
+    @ddt.unpack
+    def test_password_unicode_normalization_login(self, password, password_entered, login_success):
+        """
+        Tests unicode normalization isn't happening when flag isn't on.
+        """
+        self.user.set_password(password)
+        self.user.save()
+        response, mock_audit_log = self._login_response(self.user.email, password_entered,)
+        self._assert_response(response, success=login_success)
```

Comment:
This is identical code to the above. You can probably just merge the two tests with an additional flag in the ddt data.